src/musesounds/internal/musesoundsrepository.cpp

```diff
@@ -131,11 +131,16 @@ void MuseSoundsRepository::th_requestSounds(const UriQuery& soundsUri, std::func
         return;
     }
 
-    JsonDocument soundsInfoDoc = JsonDocument::fromJson(ByteArray::fromQByteArray(receivedData.data()));
-
     RetVal<SoundCatalogueInfoList> result;
-    result.ret = make_ret(Ret::Code::Ok);
-    result.val = parseSounds(soundsInfoDoc);
+
+    std::string err;
+    JsonDocument soundsInfoDoc = JsonDocument::fromJson(ByteArray::fromQByteArray(receivedData.data()), &err);
+    if (!err.empty()) {
+        result.ret = make_ret(static_cast<int>(Ret::Code::UnknownError), err);
```

Comment:
It would be great to add a separate error code (like Err::JsonParsingFailed)

Reply:
let's use NetworkError

Reply:
But it's not a network error...